app/controllers/events/view/export.js

```diff
@@ -0,0 +1,53 @@
+import Ember from 'ember';
+
+const { Controller, run } = Ember;
+export default Controller.extend({
+  eventExportStatus  : '',
+  eventDownloadUrl   : '',
+  isDownloadDisabled : true,
+  image              : false,
+  audio              : false,
+  video              : false,
+  document           : false,
+  actions            : {
+    startGeneration() {
+      let payload = {
+        image    : this.get('image'),
+        video    : this.get('video'),
+        audio    : this.get('audio'),
+        document : this.get('document')
+      };
+      this.get('loader')
+        .post(`/events/${this.get('model.id')}/export/json`, payload)
+        .then(exportJobInfo => {
+          let requestLoop = run.later(() => {
+            this.get('loader')
+              .load(exportJobInfo.task_url, { withoutPrefix: true })
+              .then(exportJobStatus => {
+                if (exportJobStatus.state === 'SUCCESS') {
+                  this.set('isDownloadDisabled', false);
+                  this.set('eventExportStatus', exportJobStatus.state);
+                  this.set('eventDownloadUrl', exportJobStatus.result.download_url);
+                  this.get('notify').success(this.l10n.t('Event exported.'));
+                  run.cancel(requestLoop);
+                } else if (exportJobStatus.state === 'WAITING') {
+                  this.set('eventExportStatus', exportJobStatus.state);
+                  requestLoop();
+                } else {
+                  this.set('eventExportStatus', exportJobStatus.state);
+                  run.cancel(requestLoop);
```

Comment:
And this cancelling wouldn't be needed at at all. Because `run.later` is not a loop. It is to run a piece of code **once** after a given time delay. So when this code runs, there is actually nothing to cancel since it will not run again until you ask it to run.